src/index.html

```diff
@@ -11,5 +11,15 @@
 </head>
 <body>
   <travel-app></travel-app>
+  <!--
+      What is best practice for hiding API keys in angular?
```

Comment:
Yes, the backend is where you can more effectively hide keys. Anything in JS can eventually be found which is why you may just accept that things will be seen and you can control security in different ways possibly by not allowing users to access those secure resources from anywhere. Maybe you can only access it via this URL or this server. Additionally if you do store specific data secure or not for different environments (dev/test/prod) you can do that with different environment files. That doesn't actually keep the secrets out of code though. You can keep secrets out of code by using token replacement during a build process. We aren't using a build process for this but you could be using GitHub actions or Azure DevOps pipelines and have a step that replaces config files when it sees things like %MyKey% It will replace those with keys found from a store you specify like a library or Azure Key Vault or things like that. This is a big part of DevOps that allows you to build once and deploy to many environments.